operator/v1/types_console.go

```diff
@@ -143,8 +143,97 @@ type Capability struct {
+	// The type determines whether the logo is for the dark or light theme.
+	// When set to "Dark", the corresponding logo for the dark theme will be used in the console UI.
+	// When set to "Light", the corresponding logo for the light theme will be used in the console UI.
+	// +kubebuilder:validation:Enum="Dark";"Light"
+	// +required
+	Type ThemeType `json:"type"`
+
+	// file is the reference to a specific file within a ConfigMap in the openshift-config namespace.
+	// This field allows the console to locate and use the specified file containing a custom logo.
+	// +required
+	File configv1.ConfigMapFileReference `json:"file"`
+}
+
+// CustomLogo defines a configuration based on theme types for the console UI logo.
+type CustomLogo struct {
+	// type specifies the type of the logo for the console UI. It determines whether the logo is for the masthead or favicon.
+	// This field is an enum with valid values "Masthead" and "Favicon".
+	// Valid values are "Masthead" and "Favicon".
+	// When set to "Masthead", the logo will be used in the masthead or about modal of the console UI.
+	// When set to "Favicon", the logo will be used as the favicon of the console UI.
+	// +kubebuilder:validation:Enum="Masthead";"Favicon"
+	// +required
+	Type LogoType `json:"type"`
+
+	// themes specifies the themes for the console UI logo.
+	// This field is a list of themes, each with a Type.
+	// Each theme determines whether the logo is for the dark or light theme of the console UI.
+	// The valid values for the Type field are "Dark" and "Light".
+	// The Name field should reference the name of the ConfigMap that contains the logo files.
+	// Each item in the list must have a unique combination of Type and Name.
+	// +kubebuilder:validation:Enum="Dark";"Light"
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=2
+	// +listType=map
+	// +listMapKey=type
+	// +required
+	Themes []Theme `json:"themes"`
+}
+
 // ConsoleCustomization defines a list of optional configuration for the console UI.
 type ConsoleCustomization struct {
+	// customLogos replaces the default OpenShift logo in the masthead and about dialog.
+	// It references a ConfigMap in the openshift-config namespace. You can create it with a command like:
+	// 'oc create configmap custom-logos-config \
+	//   --namespace=openshift-config \
+	//   --from-file=/path/to/file \
+	//   --from-file=/path/to/file \
+	//   --from-file=/path/to/file \
+	//   --from-file=/path/to/file'
+	// The ConfigMap keys should include file extensions for dark and light themes so that the console serves these files with the correct MIME type.
+	// Recommended Masthead and About Modal logo specifications:
+	// Image size must be less than 1 MB due to constraints on the ConfigMap size
+	// Dimensions: Max height of 68px and max width of 200px
+	// SVG format preferred
+	// Recommended Favicon specifications:
+	// Dimensions: Max height of 16px and max width of 16px
+	// PNG format preferred
```

Comment:
A couple things here:
- I think the information on the ConfigMap is likely more useful on the field that is associated with referencing the configmap itself. The command you reference could then be reduced to a single `--from-file`. 
- Explicitly mention in the godoc here that the field is optional